python/activator/middleware_interface.py

```diff
@@ -595,69 +587,66 @@ def _find_data_to_preload(self):
         if self._skip_spatial_preload:
             return (calib_datasets, calib_datasets)
 
-        detector = self.camera[self.visit.detector]
-        wcs = self._predict_wcs(detector)
-        center, radius = self._detector_bounding_circle(detector, wcs)
-
         with lsst.utils.timer.time_this(_log, msg="prep_butler (find refcats)", level=logging.DEBUG):
-            refcat_datasets = set(self._export_refcats(center, radius))
+            refcat_datasets = set(self._export_refcats(self.region))
         with lsst.utils.timer.time_this(_log, msg="prep_butler (find templates)", level=logging.DEBUG):
             template_datasets = set(self._export_skymap_and_templates(
-                center, detector, wcs, self.visit.filters))
+                self.region, self.visit.filters))
         with lsst.utils.timer.time_this(_log, msg="prep_butler (find ML models)", level=logging.DEBUG):
             model_datasets = set(self._export_ml_models())
         return (refcat_datasets | template_datasets | calib_datasets | model_datasets,
                 calib_datasets,
                 )
 
-    def _export_refcats(self, center, radius):
+    def _export_refcats(self, region):
         """Identify the refcats to export from the central butler.
 
         Parameters
         ----------
-        center : `lsst.geom.SpherePoint`
-            Center of the region to find refcat shards in.
-        radius : `lst.geom.Angle`
-            Radius to search for refcat shards in.
+        region : `lsst.sphgeom.Region`
+            The region to find refcat shards in. A circular aperture containing
+            this region is used to search all refcat shards.
 
         Returns
         -------
         refcats : iterable [`DatasetRef`]
             The refcats to be exported, after any filtering.
         """
+        center = lsst.geom.SpherePoint(region.getCentroid())
+        radius = max([center.separation(lsst.geom.SpherePoint(vertex)) for vertex in region.getVertices()])
         indexer = HtmIndexer(depth=7)
-        shard_ids, _ = indexer.getShardIds(center, radius+self.padding)
+        shard_ids, _ = indexer.getShardIds(center, radius)
```

Comment:
No with `w_2024_37` when the API first became public and when I started working on this, but yes with `w_2024_38` or newer now.
 
So, we can switch to use region query too. Thanks for motivating me to test again.

Reply:
Are we bothered that the pipelines using refcats might not be using butler in getting the shards?

Reply:
I'm not sure I understand. You mean a task might have an old-style loader and it might demand specific shards that weren't in the region?

Reply:
> You mean a task might have an old-style loader and it might demand specific shards that weren't in the region?

Honestly I don't know and have not looked enough to understand how the loader decides what shards to use.  Also, for the few examples I tried manually, the two methods (`HtmIndexer` versus butler query) give identical results.  I don't know if they always do.

Reply:
I would expect all modern tasks would get them from the Butler, I think that was part of the Gen2->3 transition.

Reply:
`_export_refcats` now no longer uses `meas.algorithms.htmIndexer` and gets overlapping refcats directly via the query in the new query system.